File: programs/hello_world_java/src/com/android/helloworldjava/HelloWorld.java
package com.android.helloworldjava;

import android.app.Activity;
import android.os.Bundle;

public class HelloWorld extends Activity
{
    @Override
    public void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main);
    }
}

File: programs/hello_world_minimal/src/main/clojure/com/android/helloworldminimal/HelloWorld.clj
(ns com.android.helloworldminimal.HelloWorld
  (:gen-class
   :extends android.app.Activity
   :exposes-methods {onCreate superOnCreate})
  (:import [android.app Activity]
           [android.os Bundle]))

(defn -onCreate [this #^android.os.Bundle bundle ]
  (.superOnCreate this bundle)
  (.setContentView this com.android.helloworldminimal.R$layout/main))
